### ISSUE TITLE
Illegal instruction & Unreachable code should be reachable

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1045]
If you try to run the following code, you will receive a `EXCEPTION_ILLEGAL_INSTRUCTION` in runtime.
Another little bug is that the last WriteLine should be reachable if `TestRet` returns a value > 0, but the IDE reports it as an unreachable code anyway.
```
using System;
using System.IO;

namespace BeefBugRepros
{
	static
	{
		public static Result<int32, int32> TestRet()
		{
			return .Ok(12);
		}

		[Test]
		public static void TestBug()
		{
			Console.WriteLine("Bug4 Begin");

			int32 a = 0;
			switch (TestRet())
			{
			case .Ok(out a):
				if (a > 0)
					break;
				fallthrough;
			case .Err(out a):
				return;
			}

			// This is reported as unreachable, but it should be reachable when a > 0.
			Console.WriteLine("Bug4 End");
		}
	}
}
```
Tested with e6fdfa3

**Comment by bfiete** [comment]
There's another bug here- `fallthrough` shouldn't be allowed when the next case has destructuring (ie- `out a` in the `.Err` case).

**Comment by bfiete** [comment]
Fixed at 5e9ac078040e9a0b085fb9ae135d00a25b8e5ae8
